### DISCUSSION TITLE
Offer auto scheduling

### DISCUSSION THREAD

**Discussion by rixx** [pretalx/pretalx#1073] [Ideas]
Now this is a big idea, but offering a first schedule draft automagically would be really, really nice.

Although this would probably escalate in a lot more functionality. We should definitely use [this library](https://github.com/PyconUK/ConferenceScheduler) here and also check out [this one](https://github.com/emfcamp/slotmachine). What is it with Python scheduling tools from the UK?

**Comment by eriktews** [comment]
As long as you are not afraid of introducing a dependency on Java, this might be interesting for the feature as well: https://www.optaplanner.org/

For example, here is one of the examples Optaplanner can solve: https://docs.optaplanner.org/7.42.0.Final/optaplanner-docs/html_single/index.html#conferenceScheduling

**Comment by rixx** [comment]
We're currently trying to avoid complex build steps, and since there are Python libraries that solve this problem already, we're probably going to stick with those. The scheduling library and algorithm is honestly not the complicated part, compared with providing a good UI for configuration and modification.
